File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/service/value/impl/AttValueDelegate.java
package ru.citeck.ecos.records3.record.op.atts.service.value.impl;

import lombok.Getter;
import org.jetbrains.annotations.NotNull;
import ru.citeck.ecos.records2.RecordRef;
import ru.citeck.ecos.records3.record.op.atts.service.value.AttEdge;
import ru.citeck.ecos.records3.record.op.atts.service.value.AttValue;

public class AttValueDelegate implements AttValue {

    @Getter
    private final AttValue impl;

    public AttValueDelegate(AttValue impl) {
        this.impl = impl == null ? EmptyAttValue.INSTANCE : impl;
    }

    @Override
    public Object getId() {
        return impl.getId();
    }

    @Override
    public String asText() throws Exception {
        return impl.asText();
    }

    @Override
    public String getDispName() throws Exception {
        return impl.getDispName();
    }

   @Override
    public Object getAtt(@NotNull String name) throws Exception {
        return impl.getAtt(name);
    }

    @Override
    public AttEdge getEdge(@NotNull String name) throws Exception {
        return impl.getEdge(name);
    }

    @Override
    public boolean has(@NotNull String name) throws Exception {
        return impl.has(name);
    }

    @Override
    public Double asDouble() throws Exception {
        return impl.asDouble();
    }

    @Override
    public Boolean asBool() throws Exception {
        return impl.asBool();
    }

    @Override
    public Object asJson() throws Exception {
        return impl.asJson();
    }

    @Override
    public Object as(@NotNull String type) throws Exception {
        return impl.as(type);
    }

    @Override
    public RecordRef getTypeRef() throws Exception {
        return impl.getTypeRef();
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/atts/value/impl/AttValueDelegate.kt
package ru.citeck.ecos.records3.record.atts.value.impl

import ru.citeck.ecos.records3.record.atts.value.AttValue

open class AttValueDelegate(val impl: AttValue) : AttValue {

    override fun init() = impl.init()
    override fun getId() = impl.id
    override fun getDisplayName() = impl.displayName
    override fun asText() = impl.asText()
    override fun getAs(type: String) = impl.getAs(type)
    override fun asDouble() = impl.asDouble()
    override fun asBoolean() = impl.asBoolean()
    override fun asJson() = impl.asJson()
    override fun asRaw() = impl.asRaw()
    override fun has(name: String) = impl.has(name)
    override fun getAtt(name: String) = impl.getAtt(name)
    override fun getEdge(name: String) = impl.getEdge(name)
    override fun getType() = impl.type
}
